lib/recipes.js

```diff
@@ -0,0 +1,40 @@
+class RecipeClass {
+  constructor(id, title, subtitle, picture) {
+    if (!id || !title || !subtitle) {
+      throw new Error("Id, title, and subtitle must be provided");
+    }
+    this.id = id;
+    this.title = title;
+    this.subtitle = subtitle;
+    this.picture = picture || "assets/recipe-placeholder.png";
+  }
+}
+
+const recipes = [
+  new RecipeClass(
+    1,
+    "Sojabällchen-Curry",
+    "Rezept aus Indien, mit Reis und Kartoffeln",
+    "assets/recipe-1.png"
+  ),
+  new RecipeClass(
+    2,
+    "Schokoladenkuchen",
+    "Vegan und ohne Mehl, einfach gemacht",
+    "assets/recipe-2.png"
+  ),
+  new RecipeClass(
+    3,
+    "Kürbis-Ingwer-Suppe",
+    "Mit Kokosmilch und Kürbiskernen",
+    "assets/recipe-3.png"
+  ),
+  new RecipeClass(
+    4,
+    "Blaubeer-Crumble",
+    "Mit Vanilleeis",
+    "assets/recipe-4.png"
+  ),
+];
+
+export { RecipeClass, recipes };
```

Comment:
code like this is not covered in our curriculum, can you explain why you chose this path?

It would be easier to use an array of objects for this.

```javascript
const recipes = [
{
  id: "1",
  name: "Crumble", 
  description: "With sauce", 
  image: "/path/of/image"
},
{...},
...
]
```

Reply:
I have defined a class and the list of recipes contains several created objects of this class, because the implementation is more structured and easier to expand. More properties will be added as my project processes, for example the working and waiting time for cooking and the cooking steps of a recipe, etc.. It is also easier to check whether required fields are filled or not. For example I want to make sure that the title and subtitle are always filled, otherwise an error message is thrown.